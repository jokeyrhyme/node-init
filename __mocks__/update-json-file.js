/* @flow */
'use strict';

let pkg /* : Object */ = {};

module.exports = async (
  pkgPath /* : string */,
  updater /* : (value: any) => Promise<any> */,
) /* : Promise<void> */ => {
  await updater(pkg);
};

module.exports.__setPackage = (p /* : Object */) => {
  pkg = p;
};
