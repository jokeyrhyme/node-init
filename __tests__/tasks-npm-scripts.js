'use strict';

jest.mock('update-json-file');

const { fn } = require('../lib/tasks/npm-scripts.js');

const TEN_SECONDS = 10 * 1e3;
jest.setTimeout(TEN_SECONDS);

test('fn() versus empty package.json', async () => {
  const pkg = {};
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() versus engines.node >= 8', async () => {
  const pkg = {
    engines: { node: '>=8' },
  };
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() with ava and nyc', async () => {
  const pkg = {
    devDependencies: {
      ava: '*',
      nyc: '*',
    },
  };
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() with ava and pre-existing scripts', async () => {
  const pkg = {
    devDependencies: {
      ava: '*',
    },
    scripts: {
      test: 'true',
    },
  };
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() versus empty package.json in Flow project', async () => {
  const pkg = {};
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path', isFlowProject: true });

  expect(pkg).toMatchSnapshot();
});

test('fn() with jest and pre-existing settings', async () => {
  const pkg = {
    devDependencies: {
      jest: '*',
    },
    jest: {
      coverageThreshold: {
        global: {
          lines: 50,
        },
      },
    },
  };
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() with postpublish', async () => {
  const pkg = {
    scripts: {
      postpublish: 'true',
    },
  };
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() with postpublish (old greenkeeper)', async () => {
  const pkg = {
    scripts: {
      postpublish: 'greenkeeper-postpublish',
    },
  };
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});
