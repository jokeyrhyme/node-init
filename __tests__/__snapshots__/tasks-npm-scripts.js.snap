// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fn() versus empty package.json 1`] = `
Object {
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "lines": 90,
      },
    },
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "eslint": "eslint --cache --fix .",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "jest": "jest",
    "lint": "npm run eslint",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "npm run jest && npm run lint",
  },
}
`;

exports[`fn() versus empty package.json in Flow project 1`] = `
Object {
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "lines": 90,
      },
    },
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "eslint": "eslint --cache --fix .",
    "flow": "flow check",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "jest": "jest",
    "lint": "npm run eslint && npm run flow",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "npm run jest && npm run lint",
  },
}
`;

exports[`fn() versus engines.node >= 8 1`] = `
Object {
  "engines": Object {
    "node": ">=8",
  },
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "lines": 90,
      },
    },
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "eslint": "eslint --cache --fix .",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "jest": "jest",
    "lint": "npm run eslint",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "npm run jest && npm run lint",
  },
}
`;

exports[`fn() with ava and nyc 1`] = `
Object {
  "devDependencies": Object {
    "ava": "*",
    "nyc": "*",
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "ava": "nyc ava",
    "eslint": "eslint --cache --fix .",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "lint": "npm run eslint",
    "nyc": "nyc check-coverage",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "npm run lint",
  },
}
`;

exports[`fn() with ava and pre-existing scripts 1`] = `
Object {
  "devDependencies": Object {
    "ava": "*",
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "ava": "ava",
    "eslint": "eslint --cache --fix .",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "lint": "npm run eslint",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "true && npm run lint",
  },
}
`;

exports[`fn() with jest and pre-existing settings 1`] = `
Object {
  "devDependencies": Object {
    "jest": "*",
  },
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "lines": 50,
      },
    },
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "eslint": "eslint --cache --fix .",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "jest": "jest",
    "lint": "npm run eslint",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "npm run jest && npm run lint",
  },
}
`;

exports[`fn() with postpublish (old greenkeeper) 1`] = `
Object {
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "lines": 90,
      },
    },
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "eslint": "eslint --cache --fix .",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "jest": "jest",
    "lint": "npm run eslint",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "npm run jest && npm run lint",
  },
}
`;

exports[`fn() with postpublish 1`] = `
Object {
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "lines": 90,
      },
    },
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "eslint": "eslint --cache --fix .",
    "fmt": "npm run sort-package-json && npm run prettier && npm run fmt:eslint",
    "fmt:eslint": "eslint --fix --no-eslintrc --parser-options \\"ecmaVersion:2018,ecmaFeatures:{jsx:true}\\" --rule \\"key-spacing:[error,{align:value}]\\" .",
    "jest": "jest",
    "lint": "npm run eslint",
    "postpublish": "true",
    "pretest": "npm run fmt",
    "prettier": "npx -q prettier --loglevel=warn --write \\"**/*.{css,js,json,jsx,md,less,scss,ts,tsx}\\"",
    "sort-package-json": "npx -q sort-package-json",
    "test": "npm run jest && npm run lint",
  },
}
`;
